### ISSUE TITLE
Feature Request: Wildcard subdomain support

### ISSUE THREAD

**Issue by enekochan** [plesk/letsencrypt-plesk#192]
Since January 4 2018 Let's Encrypt introduced a public test API endpoint for the ACME v2 protocol and wildcard support:

https://letsencrypt.org/2017/07/06/wildcard-certificates-coming-jan-2018.html
https://community.letsencrypt.org/t/acme-sh-supports-acme-v2-wildcard-now/49685

Is there any plan on adding this feature for this extension? If so, when could we expect to have it released? Thank you!

**Comment by rkosolapov** [comment]
Hi, no ETA for now.

**Comment by mrclschstr** [comment]
It's alive: https://community.letsencrypt.org/t/acme-v2-and-wildcard-certificate-support-is-live/55579

**Comment by ioweb-gr** [comment]
This would be very useful to implement. The sooner the better :)

**Comment by CHfish-ch** [comment]
I agree and hope you allow to use external name servers

**Comment by enekochan** [comment]
@rkosolapov Can this now be achieved using the command line in some way until it's supported from the Plesk control panel or using the Plesk plugin for Let's Encrypt and the command line are mutually excluding?

**Comment by mahony0** [comment]
Any hope for implementing this feature sooner?

**Comment by rkosolapov** [comment]
@enekochan you can use third-party client to get a wildcard certificate.  Plesk will not touch it (there is a little nuance - the name of the certificate should not begin with "Let's Encrypt").

Regarding the feature implementation - first version is planned to be released soon (next month maybe), that release will cover basic scenarios only.

@CHfish-ch, could you describe your case with external name servers?  Do you mean your DNS servers are not synchronised with DNS zones in Plesk?   Looks like we can't automate this case, the only thing we can do - provide an instruction what DNS record should be added to the zone.

**Comment by CHfish-ch** [comment]
@rkosolapov Indeed: My providers DNS servers are not synced with Plesk - but I'd be willing to manually add the requested entry to DNS if it is displayed... So thank you for adding such an option!

**Comment by digitall-it** [comment]
Would fix #197 when on a subdomain.

**Comment by ghost** [comment]
@rkosolapov what about it ?

**Comment by enekochan** [comment]
The final announcement of the ACMEv2 release with wildcard support: [https://community.letsencrypt.org/t/acme-v2-and-wildcard-certificate-support-is-live/55579](https://community.letsencrypt.org/t/acme-v2-and-wildcard-certificate-support-is-live/55579)

Some technical information about ACMEv2: [https://community.letsencrypt.org/t/acme-v2-production-environment-wildcards/55578](https://community.letsencrypt.org/t/acme-v2-production-environment-wildcards/55578)

**Comment by MrSnoozles** [comment]
Is this feature planned for the Plesk extension?

**Comment by Patta** [comment]
@MrSnoozles https://support.plesk.com/hc/en-us/articles/115000490174/comments/360000314893

**Comment by Patta** [comment]
It's in. Check the link above

**Comment by digitall-it** [comment]
Too bad the domains still do not renew automatically yet.

**Comment by tamer-hassan** [comment]
wildcard option still broken and the workaround instructions (link below) don't work because you can no longer uncheck the aliases options (they get checked and disabled when checking wildcard option).

https://support.plesk.com/hc/en-us/articles/360008040893-www-alias-subdomains-are-not-included-into-the-issued-wildcard-Let-s-Encrypt-Certificate

would've followed the manual instructions here:
https://talk.plesk.com/threads/lets-encrypt-wildcard-certificates.347562/

but unfortunately I have to wait a week now because I got hit by the renewal rate-limit bug described in:
https://github.com/plesk/letsencrypt-plesk/issues/207
